internal/storage/postgreSQL/treeSQL/tree.go

```diff
@@ -0,0 +1,85 @@
+package treeSQL
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"github.com/green-ecolution/green-ecolution-backend/internal/storage"
+	"github.com/green-ecolution/green-ecolution-backend/internal/storage/postgreSQL/entities/tree"
+)
+
+type TreeRepository struct {
+	db *sql.DB
+}
+
+func NewTreeRepository(db *sql.DB) *TreeRepository {
+	return &TreeRepository{db: db}
+}
+func (r *TreeRepository) Insert(ctx context.Context, data *tree.TreeEntity) error {
+	query := `
+		INSERT INTO trees (species, tree_num, age, latitude, longitude, address, additional_info)
+		VALUES ($1, $2, $3, $4, $5, $6, $7)
+	`
+	_, err := r.db.ExecContext(ctx, query, data.Species, data.TreeNum, data.Age, data.Location.Latitude, data.Location.Longitude, data.Location.Address, data.Location.AdditionalInfo)
+	if err != nil {
+		return storage.ErrCannotUpsertData
+	}
+
+	return nil
+}
+func (r *TreeRepository) Get(ctx context.Context, id string) (*tree.TreeEntity, error) {
```

Comment:
Return the domain entity
```suggestion
func (r *TreeRepository) Get(ctx context.Context, id string) (*tree.Tree, error) {
```